Library/Homebrew/exceptions.rb

```diff
@@ -191,11 +191,14 @@ def dump
 # raised by CompilerSelector if the formula fails with all of
 # the compilers available on the user's system
 class CompilerSelectionError < RuntimeError
+  attr_reader :formula
+
   def initialize(formula)
+    @formula = formula
     super <<-EOS.undent
       #{formula.name} cannot be built with any available compilers.
       To install this formula, you may need to:
-        brew install gcc
+        brew install #{formula.compiler_standards.include?(:openmp) ? "clang-omp" : "gcc"}
```

Comment:
I want avoid directly accessing data structures owned by other objects if we can help it; it makes future changes to the implementation more difficult.

Can we do something more like:

```rb
class Formula
  def needs_compiler_feature?(feature)
    active_spec.needs_compiler_feature?(feature)
  end
end


class SoftwareSpec
  def needs_compiler_feature?(feature)
    compiler_standards.include?(feature)
  end
end
```

This way we don't need to expose `compiler_standards` as part of the API.